### ISSUE TITLE
Vault-unsealer v0.2.2 can't run on microk8s arm64 m1/m2/m3 Mac or Ubuntu Arm64 server with Docker

### ISSUE THREAD

**Issue by aki-socrate** [bakito/vault-unsealer#181]
Executing command on Mac m2 with Microk8s: `kubectl run -it vault-unsealer --rm --image=ghcr.io/bakito/vault-unsealer:v0.2.2` results in an execution error:
`exec /opt/go/vault-unsealer: exec format error` typical from executing a Go program built for another arch.

Curiously, With Docker it successfully runs on same machine with command:

`docker run --platform linux/arm64 -it ghcr.io/bakito/vault-unsealer:v0.2.2 bash`

Many other multi-arch images run successfully in Microk8s on Mac without any issues (As an example: flipt, zitadel, vault, umami).

Umami and Zitadel are directly pulled from same package repo (ghcr.io) without any issue (https://github.com/umami-software/umami/pkgs/container/umami, https://github.com/zitadel/zitadel/pkgs/container/zitadel). Therefore, the issue should not relate to the package repository, but maybe to something related to the way it is built for multi architectures and then deployed to the package repo.

I have tested it with an ubuntu distrib as well on a real ARM Linux server with Docker and got the same result than with microk8s on Mac. So it does not relate to Mac only or Microk8s only.

Hope it help. Please, ask if you need more information to troubleshoot.

**Comment by aki-socrate** [comment]
Curiously, both on Mac / Microk8s  and Ubuntu 22.04 / Docker, I get this Image: "sha256:12245b9f676ca837fb68516328f0f758a1de8ac31339ccd4b62566385e7f83e7", but it does not match any referenced ImageId listed in the package repo for v0.2.2. 

I don't know to what it relates (A layer or anything else) as I can't find any ImageId Matching this sha256.

All in all, I don't know if it is relevant or not to help troubleshooting this issue.

**Comment by bakito** [comment]
Hi @akinsella-socrate 

Currently the images a built for the following 3 arch `linux/amd64,linux/arm64,linux/arm/v7` see https://github.com/bakito/vault-unsealer/blob/main/.github/workflows/publish.yml#L31

or https://github.com/bakito/vault-unsealer/pkgs/container/vault-unsealer/212243685?tag=v0.2.2 

What other arch would you expect?

I don't have the possibility to test images on Mac/arm64

**Comment by aki-socrate** [comment]
Having a look by chance on the Dockerfile while reviewing the last commit that removes `upx`execution, I can see that GOARCH is not variabilized in Dockerfile and uses value amd64.

If I am correct, Docker for Mac is able to virtualize Amd64 based binaries, while Containerd in microk8s cannot do on Mac, neither k8s can do on Linux Ubuntu. This might explain what I observed.

Variabilizing this value (as well as other potential fixed values related to arch) in the build should solve the issue and matches pretty well with the observed issue.

Here is the reference in the Dockerfile:

https://github.com/bakito/vault-unsealer/blob/dc944b9a992172a7376c1fcb6399ac391f978ec5/Dockerfile#L10

I think the parameter could be passed from here in the GHA workflow:

https://github.com/bakito/vault-unsealer/blob/dc944b9a992172a7376c1fcb6399ac391f978ec5/.github/workflows/publish.yml#L33

**Comment by bakito** [comment]
You are right. 

It should be enough to just remove the `GOARCH` env var from Dockerfile.
In branch #182 I removed the ENV variable.

The image should be pushed with tag `dockerfile.`
With build https://github.com/bakito/vault-unsealer/actions/runs/9767371397/job/26962496996

Once this is done, it would be nice if you could check it this image tag works for you.

**Comment by aki-socrate** [comment]
Removing the GOARCH as in the PR definitely fixes the issue. That's awesome !
I don't know why though, the PR build created a "dockerfile" tag in ghcr.io package repository. I assume it is not expected but I don't know from where it comes.
I tried the "dockerfile" tag on Mac microk8s and it makes it:

```shell
$ k logs vault-unsealer-f6d6cc74f-cbssx
{"level":"info","ts":"2024-07-02T20:49:08.965Z","logger":"setup","msg":"found unseal secrets","secrets":1}
{"level":"info","ts":"2024-07-02T20:49:08.971Z","logger":"setup","msg":"starting manager"}
{"level":"info","ts":"2024-07-02T20:49:08.971Z","logger":"controller-runtime.metrics","msg":"Starting metrics server"}
{"level":"info","ts":"2024-07-02T20:49:08.971Z","logger":"controller-runtime.metrics","msg":"Serving metrics server","bindAddress":":8080","secure":false}
{"level":"info","ts":"2024-07-02T20:49:08.971Z","msg":"starting server","name":"health probe","addr":"[::]:8081"}
{"level":"info","ts":"2024-07-02T20:49:08.971Z","msg":"attempting to acquire leader lease scrt-local/vault-unsealer.bakito.net..."}
{"level":"info","ts":"2024-07-02T20:49:08.977Z","msg":"successfully acquired lease scrt-local/vault-unsealer.bakito.net"}
{"level":"info","ts":"2024-07-02T20:49:08.978Z","msg":"Starting EventSource","controller":"endpoints","controllerGroup":"","controllerKind":"Endpoints","source":"kind source: *v1.Endpoints"}
{"level":"info","ts":"2024-07-02T20:49:08.978Z","msg":"Starting Controller","controller":"endpoints","controllerGroup":"","controllerKind":"Endpoints"}
{"level":"info","ts":"2024-07-02T20:49:08.978Z","msg":"Starting EventSource","controller":"pod","controllerGroup":"","controllerKind":"Pod","source":"kind source: *v1.Pod"}
{"level":"info","ts":"2024-07-02T20:49:08.978Z","msg":"Starting Controller","controller":"pod","controllerGroup":"","controllerKind":"Pod"}
{"level":"info","ts":"2024-07-02T20:49:09.082Z","msg":"Starting workers","controller":"endpoints","controllerGroup":"","controllerKind":"Endpoints","worker count":1}
{"level":"info","ts":"2024-07-02T20:49:09.082Z","msg":"Starting workers","controller":"pod","controllerGroup":"","controllerKind":"Pod","worker count":1}
{"level":"info","ts":"2024-07-02T20:49:09.130Z","logger":"vault","msg":"successfully unsealed vault","namespace":"scrt-local","pod":"vault-local-1","stateful-set":"vault-local"}
{"level":"info","ts":"2024-07-02T20:49:09.166Z","logger":"vault","msg":"successfully unsealed vault","namespace":"scrt-local","pod":"vault-local-2","stateful-set":"vault-local"}
```

It works as well in Docker on Ubuntu 22.04 on an Arm64 arch (even though it makes no direct sense in Docker to use this specific image, it validates that it will work on k8s Ubuntu 22.04 Arm64 arch). Here is the log:

```shell
~# docker run --platform linux/arm64 -it ghcr.io/bakito/vault-unsealer:dockerfile bash
Unable to find image 'ghcr.io/bakito/vault-unsealer:dockerfile' locally
dockerfile: Pulling from bakito/vault-unsealer
65cd7b79c880: Pull complete
ed7778b95764: Pull complete
Digest: sha256:c033e8289c9cc0d7ea0a1a6273b9006a0e5914e9290f442a33a5eac5d42e1769
Status: Downloaded newer image for ghcr.io/bakito/vault-unsealer:dockerfile
{"level":"error","ts":"2024-07-02T20:56:24.536Z","logger":"controller-runtime.client.config","msg":"unable to load in-cluster config","error":"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined","stacktrace":"sigs.k8s.io/controller-runtime/pkg/client/config.loadConfig.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:133\nsigs.k8s.io/controller-runtime/pkg/client/config.loadConfig\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:155\nsigs.k8s.io/controller-runtime/pkg/client/config.GetConfigWithContext\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:97\nsigs.k8s.io/controller-runtime/pkg/client/config.GetConfig\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:77\nsigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:175\nmain.main\n\t/build/main.go:53\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}
{"level":"error","ts":"2024-07-02T20:56:24.536Z","logger":"controller-runtime.client.config","msg":"unable to get kubeconfig","error":"invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable","errorCauses":[{"error":"no configuration has been provided, try setting KUBERNETES_MASTER environment variable"}],"stacktrace":"sigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/client/config/config.go:177\nmain.main\n\t/build/main.go:53\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:271"}
```

**Comment by bakito** [comment]
Thank you for the validation.

The 'dockerfile' tag is intended to be created as temporary work-around to get a tag with the fix that will be removed before merging.

**Comment by bakito** [comment]
Released with version v0.2.3
